### DISCUSSION TITLE
Where the iarc function in the run_mopac.F90 file is defined

### DISCUSSION THREAD

**Discussion by 123xyl** [openmopac/mopac#107] [Q&A]
![image](https://user-images.githubusercontent.com/96686347/185404578-79ccde5b-695d-421b-8e74-626f5e5703cb.png)

**Comment by godotalgorithm** [comment]
`iargc` and also `getarg` are common, but non-standard intrinsic Fortran subroutines that are defined by most compilers. For example, gfortran notes them in its documentation [here](https://gcc.gnu.org/onlinedocs/gcc-12.1.0/gfortran/IARGC.html). In the Fortran 2003 standard, they were declared to be standard intrinsic subroutines with the new names `command_argument_count` and `get_command_argument`, but most compilers continue to support them with either name.

Your problem suggests that either some Fortran compiler does not support these older names or else you have incorrectly installed the compiler. I'll need to know what compiler you are trying to use if you want me to be more helpful. I can switch to the newer names in MOPAC if their compiler support is better overall, or at least create an option to use either the new or old names.

**Reply by 123xyl** [reply]
flang complier

**Reply by godotalgorithm** [reply]
As I've noted in #108, I have reproduced this problem with a version of flang available on [conda-forge](https://anaconda.org/conda-forge/flang_linux-64), and I've begun to fix it. I've fixed the compiler errors, but MOPAC compiled with flang is failing 20% of my tests, and I'm still trying to fix these other problems. As long as there are no problems with the flang compiler, I should finish fixing this problem in a few days.

**Reply by 123xyl** [reply]
![image](https://user-images.githubusercontent.com/96686347/186068178-7c52a626-0e5e-497a-9bc7-10a897f08a7a.png)
This picture is my compiler, I changed it according to #108 , still did not solve the problem,

**Reply by godotalgorithm** [reply]
I think you may be a bit confused about Git and GitHub. #108 is an open Pull Request (PR). It is a change to the code in a specific Git branch (`fix-flang`) that hasn't been added to the main version of MOPAC yet. The directory path and lack of `.git` in your screenshot suggests that you downloaded a static copy of MOPAC v22.0.3. If I fix MOPAC to work with flang and merge the PR, then you will need to update your copy to the latest development version (`main` branch) to get those changes.

If you download a [snapshot](https://github.com/openmopac/mopac/archive/refs/heads/flang-fix.zip) of the open PR branch, then you can get a preview of MOPAC that should compile with flang right now, but the executable will fail on some tests and examples. If this version isn't compiling for you, then please post the compiler errors that you see. I'm working with a different version of the flang compiler, but it should be close to what you are using.

**Reply by godotalgorithm** [reply]
I finished the testing, and the main branch of MOPAC should now work with the Flang compiler. There is a compiler bug (#109) that will cause problems if you transfer input files between Windows and Linux, but it shouldn't affect normal operations of MOPAC.
